jwst/wfss_contam/wfss_contam.py

```diff
@@ -0,0 +1,88 @@
+#
+#  Top level module for WFSS contamination correction.
+#
+from os.path import splitext
+
+from jwst import datamodels
+from .observations import observation
+from ..lib.suffix import replace_suffix
+
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def contam_corr(input_model, waverange):
+    """
+    The main WFSS contam correction function
+
+    Parameters
+    ----------
+    input_model : `~jwst.datamodels.MultiSlitModel`
+        Input data model containing 2D spectral cutouts
+    waverange : `~jwst.datamodels.WavelengthrangeModel`
+        Wavelength range reference file model
+
+    Returns
+    -------
+    output_model : `~jwst.datamodels.MultiSlitModel`
+      A copy of the input_model that has been corrected
+
+    """
+    output_model = input_model.copy()
+
+    # Get the segmentation map for this grism exposure
+    seg_model = datamodels.open(input_model.meta.segmentation_map)
+
+    # Get the direct image from which the segmentation was constructed
+    direct_file = input_model.meta.direct_image
```

Comment:
The code allows for multiple direct images. We save only one in the schema. Do we expect to have (or users to supply)  multiple direct images?